src/ResourceManager/EventGrid/Commands.EventGrid/Domain/GetAzureEventGridDomain.cs

```diff
@@ -0,0 +1,128 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.EventGrid.Models;
+using Microsoft.Azure.Commands.EventGrid.Utilities;
+using Microsoft.Azure.Management.EventGrid.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.EventGrid
+{
+    /// <summary>
+    /// 'Get-AzureRmEventGridDomain' Cmdlet gives the details of a / List of EventGrid domain(s)
+    /// <para> If Domain name provided, a single Domain details will be returned</para>
+    /// <para> If Domain name not provided, list of Domains will be returned</para>
+    /// </summary>
+
+    [Cmdlet(
+        VerbsCommon.Get,
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "EventGridDomain",
+        DefaultParameterSetName = ResourceGroupNameParameterSet),
+    OutputType(typeof(PSDomain), typeof(PSDomainListInstance))]
+
+    public class GetAzureEventGridDomain : AzureEventGridCmdletBase
+    {
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
```

Comment:
@ahamad-MS nit: the `-ResourceId` parameter should be the only one in this cmdlet that has `ValueFromPipelineByPropertyName = true`

Reply:
Fixed everywhere

Reply:
@cormacpayne  Taking this back. This is actually needed here to support the following scenario

Get-AzureRmEventGridDomain -ResourceGroupName aegBugbashResourceGroup2 -Name aegBugBashDomain2 | Get-AzureRmEventGridDomainKey

Otherwise, we will get his error 
Get-AzureRmEventGridDomainKey : The input object cannot be bound to any parameters for the command either because the command does not take pipeline input or the input and its
properties do not match any of the parameters that take pipeline input